src/MongoDB.Driver/MongoClientSettings.cs

```diff
@@ -701,8 +729,14 @@ public static MongoClientSettings FromUrl(MongoUrl url)
             clientSettings.ServerSelectionTimeout = url.ServerSelectionTimeout;
             clientSettings.SocketTimeout = url.SocketTimeout;
             clientSettings.SslSettings = null; // SSL settings must be provided in code
+            clientSettings.TlsInsecure = url.TlsInsecure;
+#pragma warning disable 618
             clientSettings.UseSsl = url.UseSsl;
+#pragma warning restore 618
+            clientSettings.UseTls = url.UseTls;
+#pragma warning disable 618
             clientSettings.VerifySslCertificate = url.VerifySslCertificate;
+#pragma warning restore 618
```

Comment:
I think you should just remove the assignments for `UseSsl` and `VerifySslCertificate`. They are just aliases for `UseTls` and `TlsInsecure`.